client/extensions/woocommerce/state/sites/promotions/helpers.js

```diff
@@ -45,6 +61,32 @@ export function createPromotionFromCoupon( coupon ) {
 	return promotion;
 }
 
+export function createCouponUpdateFromPromotion( promotion ) {
+	if ( ! promotion.couponCode ) {
```

Comment:
Same here, maybe also add `! promotion ` to this clause?

Reply:
I think if there is code passing in null/undefined as a promotion here, the proper behavior is to crash, honestly.